WriteBack/ListTriggerAsync.cs

```diff
@@ -0,0 +1,85 @@
+
+namespace Microsoft.Azure.WebJobs.Extensions.Redis.Samples
+{
+    public record ListData
+    (
+        string id,
+        List<string> value
+    );
+    public static class WriteBack
+    {
+        //Redis Cache primary connection string from local.settings.json
+        public const string localhostSetting = "redisLocalhost";
+        private static readonly IDatabase cache = ConnectionMultiplexer.Connect(Environment.GetEnvironmentVariable(localhostSetting)).GetDatabase();
+
+        //CosmosDB Endpoint from local.settings.json
+        public const string Endpoint = "Endpoint";
+
+        //CosmosDB database name and container name declared here
+        public const string databaseName = "databaseName";
+        public const string containerName = "containerName";
+
+        //Uses the key of the user's choice and should be changed accordingly
+        public const string key = "listTest";
+
+        [FunctionName(nameof(ListTriggerAsync))]
+        public static async Task ListTriggerAsync(
+            [RedisListTrigger(localhostSetting, key)] string listEntry, [CosmosDB(
+            Connection = "Endpoint" )]CosmosClient client,
+            ILogger logger)
+        {
+            //Retrieve the database and container from the given client, which accesses the CosmosDB Endpoint
+            Container db = client.GetDatabase(databaseName).GetContainer(containerName);
+
+            //Creates query for item inthe container and
+            //uses feed iterator to keep track of token when receiving results from query
+            IOrderedQueryable<ListData> query = db.GetItemLinqQueryable<ListData>();
+            using FeedIterator<ListData> results = query
+                .Where(p => p.id == key)
+                .ToFeedIterator();
+
+            //Retrieve collection of items from results and then the first element of the sequence
+            var response = await results.ReadNextAsync();
+            var item = response.FirstOrDefault(defaultValue: null);
+
+            //Optional logger to display what is being pushed to CosmosDB
+            logger.LogInformation("The value added is " + listEntry);
+
+            //If there doesnt exist an entry with this key in CosmosDB, create a new entry
+            if (item == null)
+            {
+                List<string> resultsHolder = new List<string>
+                {
+                    listEntry
+                };
+
+                ListData newEntry = new ListData(id: key, value: resultsHolder);
+                await db.UpsertItemAsync(newEntry);
+            }
+
+            //If there exists an entry with this key in CosmosDB, add the new values to the existing entry
+            else
+            {
+                List<string> resultsHolder = item.value;
+
+                resultsHolder.Add(listEntry);
+                ListData newEntry = new ListData(id: key, value: resultsHolder);
+                await db.UpsertItemAsync<ListData>(newEntry);
+            }
```

Comment:
I commented this on ListTriggerSync, but it holds up here as well: I think apart from defining t List<string> resultsHolder, any other logic can be moved outside of this if/else. It would look like this: 
```
List<string> resultsHolder = new List<string>();
if (item != null)
{
    resultsHolder = item.value;
}
resultsHolder.Add(listEntry);
ListData newEntry = new ListData(id: key, value: resultsHolder);
await db.UpsertItemAsync<ListData>(newEntry);
```

Reply:
Actually, that could even be condensed by: 
```
List<string> resultsHolder = item?.value ?? new List<string>();

resultsHolder.Add(listEntry);
ListData newEntry = new ListData(id: key, value: resultsHolder);
await db.UpsertItemAsync<ListData>(newEntry);
```

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-

Reply:
@NadiamB please have a look at the second example.